src/query/service/src/pipelines/processors/transforms/hash_join/row.rs

```diff
@@ -108,7 +108,28 @@ impl RowSpace {
         }
 
         if !data_blocks.is_empty() && num_rows != 0 {
-            let data_block = DataBlock::take_blocks(&data_blocks, indices.as_slice());
+            let data_block =
+                DataBlock::take_blocks(&data_blocks, indices.as_slice(), indices.len());
+            Ok(data_block)
+        } else {
+            Ok(DataBlock::empty_with_schema(self.data_schema.clone()))
+        }
+    }
+
+    pub fn gather_build(
```

Comment:
Need to avoid duplicate during `gather_build` and `gather`

Reply:
done.